ExposedAccessKeys/lambda_functions/lookup_cloudtrail_events.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+import collections
+import boto3
+
+cloudtrail = boto3.client('cloudtrail')
+
+
+def lambda_handler(event, context):
+    account_id = event['account_id']
+    time_discovered = event['time_discovered']
+    username = event['username']
+    deleted_key = event['deleted_key']
+    exposed_location = event['exposed_location']
+    endtime = datetime.datetime.now()
+    interval = datetime.timedelta(hours=24)
+    starttime = endtime - interval
+    print('Retrieving events...')
+    events_pages = get_events_pages(username, starttime, endtime)
+    print('Summarizing events...')
+    event_names, resource_names, resource_types = get_events_summaries(events_pages)
+    return {
+        "account_id": account_id,
+        "time_discovered": time_discovered,
+        "username": username,
+        "deleted_key": deleted_key,
+        "exposed_location": exposed_location,
+        "event_names": event_names,
+        "resource_names": resource_names,
+        "resource_types": resource_types
+    }
+
+
+def get_events_pages(username, starttime, endtime):
+    try:
+        paginator = cloudtrail.get_paginator('lookup_events')
+        lookup_events_iterator = paginator.paginate(
```

Comment:
Quantity of pages may be exceedingly large if this was a heavily used username. I know you only build a summary and pagination keeps memory bounded, but the time it will take to execute may get larger than one would want. Additionally, it also opens the script up to issues with getting the next page causing a failure to get the current counter progress.

Suggestion: Add a (user configurable) limit to the number of page loads.